webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/write/PolicyWriterService.scala

```diff
@@ -581,33 +655,36 @@ class PolicyWriterServiceImpl(
       staticResource
     }
 
-    val now = System.currentTimeMillis()
-
-    val res = (parrallelSequence(staticResourceToRead) { case (templateId, agentType, templateOutPath) =>
-      for {
-        copyInfo <- techniqueRepository.getFileContent(templateId) { optInputStream =>
-          optInputStream match {
-            case None =>
-              Failure(s"Error when trying to open template '${templateId.toString}'. Check that the file exists with a ${TechniqueTemplate.templateExtension} extension and is correctly commited in Git, or that the metadata for the technique are corrects.")
-            case Some(inputStream) =>
-              logger.trace(s"Loading template: ${templateId}")
-              //string template does not allows "." in path name, so we are force to use a templateGroup by polity template (versions have . in them)
-              val content = IOUtils.toString(inputStream, StandardCharsets.UTF_8)
-              Full(TechniqueResourceCopyInfo(templateId, templateOutPath, content))
-          }
-        }
-      } yield {
-        ((copyInfo.id, agentType), copyInfo)
-      }
-    }).map( _.toMap)
-
-    logger.debug(s"${staticResourceToRead.size} techniques resources read in ${System.currentTimeMillis-now} ms")
-    res
+    for {
+      t0  <- currentTimeMillis
+      res <- (parrallelSequence(staticResourceToRead) { case (templateId, agentType, templateOutPath) =>
```

Comment:
maybe this one also should not be parralel?

Reply:
No, this one is ok, it doesn't have any parallel on top of it.